quant_queue_class.inc

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * @file
+ * QuantQueue handler.
+ */
+
+class QuantQueue implements DrupalReliableQueueInterface {
+
+  /**
+   * The name of the queue this instance is working with.
+   * 
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * Create an instance of QuantQueue.
+   */
+  public function __construct($name) {
+    $this->name = $name;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createItem($data) {
+    $db_data = serialize($data);
+    $query = db_insert('quant_queue')
+    ->fields(array(
+      'name' => $this->name,
+      'data' => $db_data,
+      'created' => time(),
+      'hash' => static::hash($this->name, $db_data),
+    ));
+
+    try {
+      return (bool) $query->execute();
+    } catch (\PDOException $error) {
+      return FALSE;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function numberOfItems() {
+    return db_query('SELECT COUNT(item_id) FROM {quant_queue} WHERE name = :name', array(
+      ':name' => $this->name,
+    ));
+  }
+
+  public function claimItem($lease_time = 3600) {
+    // Claim an item by updating its expire fields. If claim is not successful
+    // another thread may have claimed the item in the meantime. Therefore loop
+    // until an item is successfully claimed or we are reasonably sure there
+    // are no unclaimed items left.
+    while (TRUE) {
+      $item = db_query_range('SELECT data, item_id FROM {quant_queue} q WHERE expire = 0 AND name = :name ORDER BY created, item_id ASC', 0, 1, array(
+        ':name' => $this->name,
+      ))->fetchObject();
+
+      if ($item) {
+        $update = db_update('queue')
+          ->fields(array(
+              'expire' => time() + $lease_time,
+            ))
+          ->condition('item_id', $item->item_id)
+          ->condition('expire', 0);
+
+        // If there are affected rows, this update succeeded.
+        if ($update->execute()) {
```

Comment:
Couldn't this just leverage `releaseItem`?

Reply:
Not quite the same function, also this is basically a copy of https://api.drupal.org/api/drupal/modules%21system%21system.queue.inc/7.x